llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -20241,6 +20241,9 @@ static SDValue trySimplifySrlAddToRshrnb(SDValue Srl, SelectionDAG &DAG,
     return SDValue();
   unsigned ShiftValue = SrlOp1->getZExtValue();
 
+  if (ShiftValue > ResVT.getScalarSizeInBits())
```

Comment:
Is there a reason the shift amount cannot be clamped to the maximum rather than not doing the transformation?

Reply:
@paulwalker-arm I was trying to think of a case but since the add is check for single use and the top half of the elements get narrowed I think this it's safe to clamp it to the maximum value.